### PR TITLE
Feat/Explorer-34 - Add block details page

### PR DESCRIPTION
**Pull request by VladislavPopovSR**
https://app.zenhub.com/workspaces/iroha-v2-60ddb820813b9100181fc060/issues/gh/soramitsu/iroha2-block-explorer-web/34

### CODE REVIEW

File: src/shared/api/index.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ interface HTTPService {
   fetchPeerStatus: () => Promise<Status>
   fetchRoles: () => Promise<Role[]>
   fetchBlocks: (params?: PaginationParams) => Promise<Paginated<BlockShallow>>
+  fetchBlock: (height: number) => Promise<BlockShallow>
```
Comment (0x009922) [inline src/shared/api/index.ts:14]:
```suggestion
  fetchBlock: (heightOrHash: number | string) => Promise<BlockShallow>
```

File: src/shared/api/http.ts [hunk 2]
```diff
@@ -59,6 +61,11 @@ export function fetchBlocks(params?: PaginationParams): Promise<Paginated<BlockS
   return get('/blocks', params);
 }
 
+export async function fetchBlock(height: number): Promise<BlockShallow> {
+  // FIXME: replace with real api
+  return makeBlock();
+}
```
Comment (0x009922) [inline src/shared/api/http.ts:67]: Mocks are used in fake http.

```suggestion
export async function fetchBlock(heightOrHash: number | string): Promise<BlockShallow> {
  return get(`/blocks/${heightOrHash}`)
}
```

File: src/pages/Blocks/BlockDetailsPage.vue [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+const { width } = useWindowSize();
+
+const transactionHashType = computed(() => (width.value < TRANSACTION_HASH_BREAKPOINT ? 'short' : 'full'));
+
+const blockHeight = computed(() => {
+  const height = router.currentRoute.value.params['height'];
+
+  if (typeof height === 'string') return height;
+
+  return height[0];
```
Comment (0x009922) [inline src/pages/Blocks/BlockDetailsPage.vue:32]: It should never be an array. Use `invariant` to assert that the parameter should be a single string, the only acceptable value.

File: src/pages/Blocks/BlockDetailsPage.vue [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+
+const { handleUnknownError } = useErrorHandlers();
+
+const TRANSACTION_HASH_BREAKPOINT = 1200;
+const { width } = useWindowSize();
+
+const transactionHashType = computed(() => (width.value < TRANSACTION_HASH_BREAKPOINT ? 'short' : 'full'));
+
+const blockHeight = computed(() => {
+  const height = router.currentRoute.value.params['height'];
```
Comment (0x009922) [inline src/pages/Blocks/BlockDetailsPage.vue:28]: It is height or hash.

File: src/app/router.ts [hunk 5]
```diff
@@ -9,8 +9,14 @@ const routes = [
   {
     path: '/blocks',
     name: 'blocks-list',
-    component: () => import('@/pages/BlocksList.vue'),
+    component: () => import('@/pages/Blocks/BlocksList.vue'),
   },
+  {
+    path: '/blocks/:height',
```
Comment (0x009922) [inline src/app/router.ts:15]:
```suggestion
    path: '/blocks/:heightOrHash',
```

**Review by 0x009922 — COMMENTED** [review]
Fine, in general.

BTW I am not sure that creating nested directories in `pages` is needed. Flat structure is always better than nested (Python's Zen!) when there is no strong reason against it.

**Comment by VladislavPopovSR** [comment]
I would keep nested directories since they could contain consts types and so on related to page. Also, is it correct expression for previous block arrow display?
```
isPreviousBlockExists.value = block.value.height > 1;
```

**Comment by 0x009922** [comment]
> Also, is it correct expression for previous block arrow display?

Yes, it is.



> I would keep nested directories since they could contain consts types and so on related to the page

Those are to be located in related "features" directory, aren't they?

File: src/pages/AccountDetailsPage.vue [hunk 6]
```diff
@@ -14,6 +14,7 @@ import { useWindowSize } from '@vueuse/core';
 import { format } from '@/shared/lib/time';
 import BaseLoading from '@/shared/ui/components/BaseLoading.vue';
 import { useErrorHandlers } from '@/shared/ui/composables/useErrorHandlers';
+import invariant from 'ts-invariant';
```
Comment (0x009922) [inline src/pages/AccountDetailsPage.vue:17]: `tiny-invariant` is a more standard one.

**Review by 0x009922 — APPROVED** [review]

**Pull request merged**